react/features/filmstrip/components/web/Thumbnail.js

```diff
@@ -182,14 +278,86 @@ class Thumbnail extends Component<Props, State> {
      * @inheritdoc
      * @returns {void}
      */
-    componentDidUpdate(prevProps: Props) {
+    componentDidUpdate(prevProps: Props, prevState: State) {
         if (prevProps._audioTrack !== this.props._audioTrack) {
             this._stopListeningForAudioUpdates(prevProps._audioTrack);
             this._listenForAudioUpdates();
             this._updateAudioLevel(0);
         }
+
+        if (prevState.displayMode !== this.state.displayMode) {
+            this._onDisplayModeChanged();
+        }
+    }
+
+    /**
+     * Handles display mode changes.
+     *
+     * @returns {void}
+     */
+    _onDisplayModeChanged() {
+        const input = getDisplayModeInput(this.props, this.state);
+        const displayModeString = DISPLAY_MODE_TO_STRING[this.state.displayMode];
+        const id = this.props._participant?.id;
+
+
+        this._maybeSendScreenSharingIssueEvents(input);
+        logger.debug(`Displaying ${displayModeString} for ${id}, data: [${JSON.stringify(input)}]`);
+    }
+
+    /**
+     * Sends screen sharing issue event if an issue is detected.
+     *
+     * @param {Object} input - The input used to compute the thumbnail display mode.
+     * @returns {void}
+     */
+    _maybeSendScreenSharingIssueEvents(input) {
+        const {
+            _currentLayout,
+            _isAudioOnly,
+            _isScreenSharing
+        } = this.props;
+        const { displayMode } = this.state;
+        const tileViewActive = _currentLayout === LAYOUTS.TILE_VIEW;
+
+        if (displayMode !== DISPLAY_VIDEO
```

Comment:
Maybe this reads better? `![ DISPLAY_VIDEO, DISPLAY_VIDEO_WITH_NAME ].includes(displayMode)` ?

Reply:
Yep!